### ISSUE TITLE
[Rule Tuning] Abnormal Process ID or Lock File Created

### ISSUE THREAD

**Issue by BenB196** [elastic/detection-rules#2979]
## Link to rule

https://github.com/elastic/detection-rules/blob/main/rules/linux/execution_abnormal_process_id_file_created.toml

## Description

There are a few things with this rule that I think can be improved on.

### Inefficient `file.path : (/var/run/* or /run/*)`

This seems to be extremely inefficient.

On a 9-minute window it takes several seconds to complete:

![image](https://github.com/elastic/detection-rules/assets/8277432/93894313-9e47-4e24-968a-96ce263b4bc6)

If you change this part to: `file.path.text : ("/var/run/" or "/run/")`

On the same 9-minute window, it now takes <1 second to complete:

![image](https://github.com/elastic/detection-rules/assets/8277432/5767b367-178a-4139-a61f-51259f1ed005)

### Wildcards wrapped in double quotes

In this part of the query

```
not 
file.name : ("auditd.pid" or "python*" or "apport.pid" or "apport.lock" or "kworker*" or "gdm3.pid" or "sshd.pid" or 
"acpid.pid" or "unattended-upgrades.lock" or "unattended-upgrades.pid" or "cmd.pid" or "yum.pid" or "netconfig.pid" or 
"docker.pid" or "atd.pid" or "lfd.pid" or "atop.pid" or "nginx.pid" or "dhclient.pid" or "smtpd.pid" or "stunnel.pid" or 
"1_waagent.pid" or "crond.pid" or "cron.reboot" or "sssd.pid" or "tomcat8.pid")
```

There are few places where wildcards are used, but are wrapped in double quotes, making them literal strings and therefore not doing what they're intended to do.

- `"python*"`
- `"kworker*"`

### `/run/` and Kubernetes

In a number of Kubernetes distributions, `/run/` is used for containerd (ex: RKE2/K3S)

`/run/k3s/containerd/io.containerd.runtime.v2.task/k8s.io/0f3753f3a7db0d342551aaaa07448fba0e589fcd0964abfe71385255c9a7856a/.f3babd79a993c3dd20b13670a332856173d17562ef6a06233a49ffc046a90ab9.pid`

A potential solution here would be to convert:

`and file.path.text : ("/var/run/" or "/run/")`

into something like:

`and (file.path.text : ("/var/run/") or (file.path.text : "/run/" and not file.path.text : "/containerd/"))`

I'm not 100% sure the implications of this addition though within the context/intent of this rule.

### Additional Common Files

The rule already excludes a lot of common file names, I'd recommend adding the following additional exclusions:

- `winbindd.pid`
    - Generally found under something like: `/run/samba/winbindd.pid`
- `chronyd.pid`
    - Generally found under something like: `/var/run/chrony/chronyd.pid`

**Comment by BenB196** [comment]
Hi @Aegrah regarding the wildcard:

I believe something that might have been missed, was me suggesting changing `file.path` -> `file.path.text`.

If you make the above field change, and then use the non-wildcard paths, in theory, the rule _should_ still be able to detect things like:

`/var/run/asdf/...`
&
`/run/asdf/...`

Note: I think it would also technically be able to match some other things like:

`/asdf/var/run/...`

But these would be pretty weird paths to have I'd think, so I'm not sure how big of a deal that would be.

(I'm also not sure Elastic's stance on using `.text` sub-fields in rules.

**Comment by Aegrah** [comment]
Hi @BenB196. We try to avoid querying .text fields as much as possible. I did however try it out, just to see how it would perform, and I what I noticed were some additional FPs like you mentioned in your example.

When changing the query to `file.path.text : ("/var/run/" or "/run/")` additional hits that contain `/run/` anywhere in its path such as e.g. `/opt/Elastic/Agent/data/elastic-agent/run/osquery-default/osquery/osquery.pid` are returned.  

Your approach definitely would work - and feel free to change the query in your environment to reflect your tunings, as it defenitly does the job, good find - however we tend to not query .text fields unless deemed entirely necessary. Exceptions to this would be for example powershell script blocks, as those are returned in text.
